### PR TITLE
bump to vale 3

### LINKED ISSUE #22171: Repo Tooling: Switch to use Vale 3.0
### 🔖 Feature description

Vale 3.0 has just been [released](https://github.com/errata-ai/vale/releases/tag/v3.0.0). We should migrate this repo to use Value 3.0 instead, since we are currently using v2.

Appreciate any help with this effort! 🙏 

### 🎤 Context

#22172

### ✌️ Possible Implementation

The following needs to be considered in the migration:

- The configuration at `.vale.ini`
- The configuration in `.github/vale`
- The docs quality workflow at `.github/workflows/verify_docs-quality.yml`
- The script that generates arguments for the above at `scripts/check-docs-quality.js`

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by freben**
Closes #22171

https://github.com/errata-ai/vale/releases/tag/v3.0.0

> [!NOTE]
> This requires all contributors to upgrade to vale 3 locally too, since the vocab files have changed locations.

I considered making a softlink or keeping duplicates or whatnot, but it feels like it'll just add risk over time instead of ripping off the band aid.

### CODE REVIEW

File: .github/workflows/verify_docs-quality.yml [hunk 1]
```diff
@@ -30,6 +30,6 @@ jobs:
         with:
           # This also contains --config=.github/vale/config.ini ... :/
           files: '${{ steps.generate.outputs.args }}'
-          version: 2.30.0 # TODO: migrate to v3
+          version: 3.0.4
```
Comment (freben) [inline .github/workflows/verify_docs-quality.yml:33]: I guess this is a point of interest. Whether we should pin the version to not have to worry about major shifts in the future, but to also lose out on updates to vale. Which is worse? Maybe we can assume that it'll take a long time before the next major that has a breakage that actually affects our current setup?

File: .github/workflows/verify_docs-quality.yml [hunk 2]
```diff
@@ -30,6 +30,6 @@ jobs:
         with:
           # This also contains --config=.github/vale/config.ini ... :/
           files: '${{ steps.generate.outputs.args }}'
-          version: 2.30.0 # TODO: migrate to v3
+          version: latest
```
Comment (freben) [inline .github/workflows/verify_docs-quality.yml:33]: For the reviewer: I went a bit back and forth on what to put here. You can't do like `^3` or so, it has to be exact versions or `latest` specifically.

Putting a fixed version like `3.0.2` means no security updates and patches. Putting `latest` means we may see some future errors when version 4 is released.

In the end I thought this is the least bad option at this point.

Comment (Rugvip) [reply .github/workflows/verify_docs-quality.yml:33]: Yep 👍

**Review by Rugvip — APPROVED** [review]
Nice! 👍

**Pull request merged**